parser/ast/procedure_test.go

```diff
@@ -0,0 +1,149 @@
+			"select *,u.username,ua.address,CONCAT(u.username, \"-\" ,ua.address) as userinfo,avg(us.score) over (order by us.id rows 2 preceding) as current_avg,sum(score) over (order by us.id rows 2 preceding) as current_sum from user_score us" +
+			" left join user u on u.id = us.user_id left join user_address ua on ua.id = us.user_id;" +
+			"select a.id,a.username,a.password,a.age,a.sex from user a where a.id in (select user_id from user_score where score > 90);	end;",
+		"create procedure proc_2() begin select us.user_id,u.username,us.subject,us.score from user_score us left join user u on u.id = us.user_id where us.score > 90 group by us.user_id,us.subject,us.score;" +
+			"select us.user_id,u.username,us.subject,us.score from user_score us join user u on u.id = us.user_id where us.score > 90 group by us.user_id,us.subject,us.score;" +
+			"select a.id,a.username,a.password,a.age,a.sex,ad.address,CONCAT(a.username, \"-\" ,ad.address) as userinfo from user a left join user_address ad on a.id = ad.user_id where a.id > 10 and a.id < 50;" +
+			"select a.id,a.username,a.password,a.age,a.sex,ad.score from user a right join user_score ad on a.id = ad.user_id where a.id > 10 and a.id < 50;" +
+			"select a.id,a.username,a.password,a.age,a.sex,ad.score from user a left join user_score ad on a.id = ad.user_id where a.id in (select user_id from user_score where score > 90 and score < 99 ) " +
+			"union select a.id,a.username,a.password,a.age,a.sex,ad.score from user a left join user_score ad on a.id = ad.user_id where a.id in (select user_id from user_score where score > 30 and score < 70 ); end;",
+		`create procedure proc_2() begin select * from t1; if i > 1 then select 2 ; end if ;end`,
+		`create procedure proc_2() begin select * from t1; if i > 1 then select 2; else select id from t2  ; end if ; end`,
+		`create procedure proc_2() begin select * from t1; if i > 1 then select 2; elseif i = 3 then select 4; else select 5 ; end if; end`,
+		`create procedure proc_2(id int) begin while id < 10 do set id = id + 1; select 1; end while; end`,
+		`create procedure proc_2() begin declare test1 CURSOR for select 1; end;`,
+		`create procedure proc_2() begin declare test1 CURSOR for select 1; select 1;open test1; end;`,
+		`create procedure proc_2() begin declare a int;declare test1 CURSOR for select 1; select 1;open test1; fetch test1 into a; close test1;end;`,
+		`create procedure proc_2() begin declare a int;declare exit handler for 1111 select 1 ; end;`,
+		`create procedure proc_2() begin declare a int;declare exit handler for 1111,1112 select 1 ; end;`,
+		`create procedure proc_2() begin declare a int;declare exit handler for SQLWARNING,NOT FOUND,SQLEXCEPTION select 1 ; end;`,
+		`create procedure proc_2() begin declare a int;declare continue handler for SQLWARNING,NOT FOUND,SQLEXCEPTION select 1 ; end;`,
```

Comment:
what's the result for executing `create procedure proc_2()....`?  error , waning, or nothing?

Reply:
Successfully converted to ast tree

Reply:
So it will return success to client?